skfda/ml/regression/_fpls_regression.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Any, TypeVar, Union
+
+from sklearn.utils.validation import check_is_fitted
+
+from ..._utils._sklearn_adapter import BaseEstimator, RegressorMixin
+from ...misc.regularization import L2Regularization
+from ...preprocessing.dim_reduction import FPLS
+from ...representation import FDataGrid
+from ...representation.basis import Basis, FDataBasis
+from ...typing._numpy import NDArrayFloat
+
+InputType = TypeVar(
+    "InputType",
+    bound=Union[FDataGrid, FDataBasis, NDArrayFloat],
+)
+
+OutputType = TypeVar(
+    "OutputType",
+    bound=Union[FDataGrid, FDataBasis, NDArrayFloat],
+)
+
+
+class FPLSRegression(
+    BaseEstimator,
+    RegressorMixin[InputType, OutputType],
+):
+    r"""
+    Regression using Functional Partial Least Squares.
+
+    Parameters:
+        n_components: Number of components to keep. Defaults to 5.
+        regularization_X: Regularization for the calculation of the X weights.
+        weight_basis_X: Basis to use for the X block. Only
+            applicable if X is a FDataBasis. Otherwise it must be None.
+        weight_basis_Y: Basis to use for the Y block. Only
+            applicable if Y is a FDataBasis. Otherwise it must be None.
+        _integration_weights_X: One-dimensional array with the integration
```

Comment:
This is not a public parameter (and should not be a parameter IMHO). Do not document it in the docstring.

Reply:
It's needed for the tests. In particular, for the comparisons with `fda.usc` and `sklearn`. I'll remove it from the documentation.

Reply:
I do not think so.